Index: git/rdkb_log4crc
===================================================================
--- git.orig/rdkb_log4crc
+++ git/rdkb_log4crc
@@ -111,6 +111,7 @@
     <rollingpolicy name="CPUPROCANALYZER_rollingpolicy" type="sizewin" maxsize="2097152" maxnum="2"/>
     <rollingpolicy name="HWST_rollingpolicy" type="sizewin" maxsize="2097152" maxnum="2"/>
     <rollingpolicy name="OVSAGENT_rollingpolicy" type="sizewin" maxsize="2097152" maxnum="2"/>
+    <rollingpolicy name="INANGOPLUG_rollingpolicy" type="sizewin" maxsize="2097152" maxnum="2"/>
     <rollingpolicy name="WANMANAGER_rollingpolicy" type="sizewin" maxsize="2097152" maxnum="2"/>
     <rollingpolicy name="FWUPGRADEMANAGER_rollingpolicy" type="sizewin" maxsize="2097152" maxnum="2"/>
 
@@ -167,6 +168,7 @@
     <appender name="RI_CPUPROCANALYZERrollingfileappender" type="rollingfile" logdir="/rdklogs/logs/" prefix="CPUPROCANALYZERlog.txt" layout="comcast_dated" rollingpolicy="CPUPROCANALYZER_rollingpolicy"/>
     <appender name="RI_HWSTrollingfileappender" type="rollingfile" logdir="/rdklogs/logs/" prefix="HWSTLog.txt" layout="comcast_dated" rollingpolicy="HWST_rollingpolicy"/>
     <appender name="RI_OVSAGENTrollingfileappender" type="rollingfile" logdir="/rdklogs/logs/" prefix="OvsAgentLog.txt" layout="comcast_dated" rollingpolicy="OVSAGENT_rollingpolicy"/>
+    <appender name="RI_INANGOPLUGrollingfileappender" type="rollingfile" logdir="/rdklogs/logs/" prefix="InangoplugLog.txt" layout="comcast_dated" rollingpolicy="INANGOPLUG_rollingpolicy"/>
     <appender name="RI_WANMANAGERrollingfileappender" type="rollingfile" logdir="/rdklogs/logs/" prefix="WANMANAGERLog.txt" layout="comcast_dated" rollingpolicy="WANMANAGER_rollingpolicy"/>
     <appender name="RI_FWUPGRADEMANAGERrollingfileappender" type="rollingfile" logdir="/rdklogs/logs/" prefix="FwUpgradeManagerLog.txt" layout="comcast_dated" rollingpolicy="FWUPGRADEMANAGER_rollingpolicy"/>
     <appender name="stdout" type="stream_env" layout="comcast_dated"/>
@@ -254,6 +256,7 @@
     <category name="RI.CPUPROCANALYZER" priority="debug" appender="RI_CPUPROCANALYZERrollingfileappender"/>
     <category name="RI.HWST" priority="debug" appender="RI_HWSTrollingfileappender"/>
     <category name="RI.OVSAGENT" priority="debug" appender="RI_OVSAGENTrollingfileappender"/>
+    <category name="RI.INANGOPLUG" priority="debug" appender="RI_INANGOPLUGrollingfileappender"/>
     <category name="RI.WANMANAGER" priority="debug" appender="RI_WANMANAGERrollingfileappender"/>
     <category name="RI.FWUPGRADEMANAGER" priority="debug" appender="RI_FWUPGRADEMANAGERrollingfileappender"/>
 
@@ -404,6 +407,8 @@
 
      <category name="RI.Stack.OVSAGENT" priority="debug" appender="RI_OVSAGENTrollingfileappender"/>
      <category name="RI.Stack.LOG.RDK.OVSAGENT" priority="debug" appender="RI_OVSAGENTrollingfileappender"/>
+     <category name="RI.Stack.INANGOPLUG" priority="debug" appender="RI_INANGOPLUGrollingfileappender"/>
+     <category name="RI.Stack.LOG.RDK.INANGOPLUG" priority="debug" appender="RI_INANGOPLUGrollingfileappender"/>
      <category name="RI.Stack.WANMANAGER" priority="debug" appender="RI_WANMANAGERrollingfileappender"/>
      <category name="RI.Stack.LOG.RDK.WANMANAGER" priority="debug" appender="RI_WANMANAGERrollingfileappender"/>	
      <category name="RI.Stack.FWUPGRADEMANAGER" priority="debug" appender="RI_FWUPGRADEMANAGERrollingfileappender"/>
Index: git/rdkb_debug.ini
===================================================================
--- git.orig/rdkb_debug.ini
+++ git/rdkb_debug.ini
@@ -72,6 +72,7 @@ LOG.RDK.XTMAGENT = FATAL ERROR WARNING N
 LOG.RDK.CPUPROCANALYZER = FATAL ERROR WARNING NOTICE INFO
 LOG.RDK.HWST = FATAL ERROR WARNING NOTICE INFO
 LOG.RDK.OVSAGENT = FATAL ERROR WARNING NOTICE INFO
+LOG.RDK.INANGOPLUG = FATAL ERROR WARNING NOTICE INFO
 LOG.RDK.WANMANAGER = FATAL ERROR WARNING NOTICE INFO
 LOG.RDK.FWUPGRADEMANAGER = FATAL ERROR WARNING NOTICE INFO
 
