Index: git/rdkb_debug.ini
===================================================================
--- git.orig/rdkb_debug.ini
+++ git/rdkb_debug.ini
@@ -62,6 +62,7 @@ LOG.RDK.SPDTST = ALL FATAL ERROR WARNING
 LOG.RDK.BLE = ALL FATAL ERROR WARNING NOTICE INFO DEBUG TRACE
 LOG.RDK.WANAGENT = ALL FATAL ERROR WARNING NOTICE INFO DEBUG TRACE
 LOG.RDK.TELCOVOIPAGENT = ALL FATAL ERROR WARNING NOTICE INFO DEBUG TRACE
+LOG.RDK.INANGOPLUG = ALL FATAL ERROR WARNING NOTICE INFO DEBUG TRACE
 LOG.RDK.WEBCONFIG = ALL FATAL ERROR WARNING NOTICE INFO DEBUG TRACE
 LOG.RDK.ADVSEC = ALL FATAL ERROR WARNING NOTICE INFO DEBUG TRACE
 LOG.RDK.XDNS = ALL FATAL ERROR WARNING NOTICE INFO DEBUG TRACE
Index: git/rdkb_log4crc
===================================================================
--- git.orig/rdkb_log4crc
+++ git/rdkb_log4crc
@@ -105,6 +105,7 @@
     <rollingpolicy name="BLE_rollingpolicy" type="sizewin" maxsize="2097152" maxnum="2"/>
     <rollingpolicy name="WANAGENT_rollingpolicy" type="sizewin" maxsize="2097152" maxnum="2"/>
     <rollingpolicy name="TELCOVOIPAGENT_rollingpolicy" type="sizewin" maxsize="2097152" maxnum="2"/>
+    <rollingpolicy name="INANGOPLUG_rollingpolicy" type="sizewin" maxsize="2097152" maxnum="2"/>
     <rollingpolicy name="WEBCONFIG_rollingpolicy" type="sizewin" maxsize="2097152" maxnum="2"/>
     <rollingpolicy name="XDNS_rollingpolicy" type="sizewin" maxsize="2097152" maxnum="2"/>
     <rollingpolicy name="DSLAGENT_rollingpolicy" type="sizewin" maxsize="2097152" maxnum="2"/>
@@ -164,6 +165,7 @@
     <appender name="RI_WEBCONFIGrollingfileappender" type="rollingfile" logdir="/rdklogs/logs/" prefix="WEBCONFIGlog.txt" layout="comcast_dated" rollingpolicy="WEBCONFIG_rollingpolicy"/>
     <appender name="RI_XDNSrollingfileappender" type="rollingfile" logdir="/rdklogs/logs/" prefix="XDNSlog.txt" layout="comcast_dated" rollingpolicy="XDNS_rollingpolicy"/>
     <appender name="RI_DSLAGENTrollingfileappender" type="rollingfile" logdir="/rdklogs/logs/" prefix="DSLAGENTLog.txt" layout="comcast_dated" rollingpolicy="DSLAGENT_rollingpolicy"/>
+    <appender name="RI_INANGOPLUGrollingfileappender" type="rollingfile" logdir="/rdklogs/logs/" prefix="InangoplugLog.txt" layout="comcast_dated" rollingpolicy="INANGOPLUG_rollingpolicy"/>
     <appender name="RI_VLANAGENTrollingfileappender" type="rollingfile" logdir="/rdklogs/logs/" prefix="VLANAGENTLog.txt" layout="comcast_dated" rollingpolicy="VLANAGENT_rollingpolicy"/>
     <appender name="RI_XTMAGENTrollingfileappender" type="rollingfile" logdir="/rdklogs/logs/" prefix="XTMAGENTLog.txt" layout="comcast_dated" rollingpolicy="XTMAGENT_rollingpolicy"/>
     <appender name="RI_CPUPROCANALYZERrollingfileappender" type="rollingfile" logdir="/rdklogs/logs/" prefix="CPUPROCANALYZERlog.txt" layout="comcast_dated" rollingpolicy="CPUPROCANALYZER_rollingpolicy"/>
@@ -249,6 +251,7 @@
     <category name="RI.BLE" priority="debug" appender="RI_BLErollingfileappender"/>
     <category name="RI.WANAGENT" priority="debug" appender="RI_WANAGENTrollingfileappender"/>
     <category name="RI.TELCOVOIPAGENT" priority="debug" appender="RI_TELCOVOIPAGENTrollingfileappender"/>
+    <category name="RI.INANGOPLUG" priority="debug" appender="RI_INANGOPLUGrollingfileappender"/>
     <category name="RI.WEBCONFIG" priority="debug" appender="RI_WEBCONFIGrollingfileappender"/>
     <category name="RI.XDNS" priority="debug" appender="RI_XDNSrollingfileappender"/>
     <category name="RI.DSLAGENT" priority="debug" appender="RI_DSLAGENTrollingfileappender"/>
@@ -395,6 +398,8 @@
 
      <category name="RI.Stack.BLE" priority="debug" appender="RI_BLErollingfileappender"/>
         <category name="RI.Stack.LOG.RDK.BLE" priority="debug" appender="RI_BLErollingfileappender"/>
+     <category name="RI.Stack.INANGOPLUG" priority="debug" appender="RI_INANGOPLUGrollingfileappender"/>
+     <category name="RI.Stack.LOG.RDK.INANGOPLUG" priority="debug" appender="RI_INANGOPLUGrollingfileappender"/>
 
      <category name="RI.Stack.WEBCONFIG" priority="debug" appender="RI_WEBCONFIGrollingfileappender"/>
      <category name="RI.Stack.LOG.RDK.WEBCONFIG" priority="debug" appender="RI_WEBCONFIGrollingfileappender"/>
