From b9a8036b2ac04b7e8580fbcb31c2625c87b7bb6d Mon Sep 17 00:00:00 2001
From: Anton Chigrin <a.chigrin@inango-systems.com>
Date: Fri, 11 Feb 2022 19:25:34 +0500
Subject: [PATCH] ioctl fixes

---
 datapath/brcompat_main.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/datapath/brcompat_main.c b/datapath/brcompat_main.c
index ac0c05c2d..6a96fc491 100644
--- a/datapath/brcompat_main.c
+++ b/datapath/brcompat_main.c
@@ -797,7 +797,7 @@ static int old_dev_ioctl(struct net_device *dev, struct ifreq *rq, int cmd)
 }
 
 /* Called with the rtnl_lock. */
-static int brc_dev_ioctl(struct net_device *dev, struct ifreq *rq, int cmd)
+static int brc_dev_ioctl(struct net_device *dev, struct ifreq *rq, void __user *data, int cmd)
 {
 	int err;
 
