diff --git a/src/gw_control/src/prov/src/gw_api_proxy/include/gw_api_proxy.h b/src/gw_control/src/prov/src/gw_api_proxy/include/gw_api_proxy.h
index 57958754..99bbabbc 100755
--- a/src/gw_control/src/prov/src/gw_api_proxy/include/gw_api_proxy.h
+++ b/src/gw_control/src/prov/src/gw_api_proxy/include/gw_api_proxy.h
@@ -712,6 +712,9 @@ extern  bool_t proxy_cm_hal_gw_mode_change_1_svc(avalanche_pp_gw_mode_e , PROXY_
 #define PROXY_docsis_GetDsOfdmChanTable 1212
 extern  enum clnt_stat proxy_docsis_getdsofdmchantable_1(RPC_PROXY_DOCSIF31_CM_DS_OFDM_CHAN *, CLIENT *);
 extern  bool_t proxy_docsis_getdsofdmchantable_1_svc(RPC_PROXY_DOCSIF31_CM_DS_OFDM_CHAN *, struct svc_req *);
+#define PROXY_SET_ARM_INANGOPLUG_ENABLE 1213
+extern  enum clnt_stat proxy_set_arm_inangoplug_enable_1(PROXY_STATUS *, CLIENT *);
+extern  bool_t proxy_set_arm_inangoplug_enable_1_svc(PROXY_STATUS *, struct svc_req *);
 
 extern int cgm_gw_api_proxy_prog_number_1_freeresult (SVCXPRT *, xdrproc_t, caddr_t);
 
@@ -945,6 +948,9 @@ extern  bool_t proxy_cm_hal_gw_mode_change_1_svc();
 #define PROXY_docsis_GetDsOfdmChanTable 1212
 extern  enum clnt_stat proxy_docsis_getdsofdmchantable_1();
 extern  bool_t proxy_docsis_getdsofdmchantable_1_svc();
+#define PROXY_SET_ARM_INANGOPLUG_ENABLE 1213
+extern  enum clnt_stat proxy_set_arm_inangoplug_enable_1();
+extern  bool_t proxy_set_arm_inangoplug_enable_1_svc();
 
 extern int cgm_gw_api_proxy_prog_number_1_freeresult ();
 #endif /* K&R C */
diff --git a/src/gw_control/src/prov/src/gw_api_proxy/src/gw_api_proxy_server.c b/src/gw_control/src/prov/src/gw_api_proxy/src/gw_api_proxy_server.c
index dc965953..8e9437f5 100755
--- a/src/gw_control/src/prov/src/gw_api_proxy/src/gw_api_proxy_server.c
+++ b/src/gw_control/src/prov/src/gw_api_proxy/src/gw_api_proxy_server.c
@@ -5630,6 +5630,22 @@ proxy_cm_hal_gw_mode_change_1_svc(avalanche_pp_gw_mode_e arg1, PROXY_STATUS *res
 	return retval;
 }
 
+bool_t
+proxy_set_arm_inangoplug_enable_1_svc(PROXY_STATUS *result, struct svc_req *rqstp)
+{
+	bool_t retval = TRUE;
+	FILE * file;
+
+	file = fopen("/tmp/inangoplug_enable", "a");
+	if (file == NULL) {
+		LOG_GW_ERROR(" Failed to open /tmp/inangoplug_enable.\n");
+		return retval;
+	}
+	fclose(file);
+	return retval;
+
+}
+
 /********************************************************************************************/
 /*! \fn bool_t   cgm_gw_api_proxy_prog_number_1_freeresult (SVCXPRT *transp, xdrproc_t xdr_result, caddr_t result)
  *********************************************************************************************
diff --git a/src/gw_control/src/prov/src/gw_api_proxy/src/gw_api_proxy_svc.c b/src/gw_control/src/prov/src/gw_api_proxy/src/gw_api_proxy_svc.c
index ff664e5a..73ff1a5d 100644
--- a/src/gw_control/src/prov/src/gw_api_proxy/src/gw_api_proxy_svc.c
+++ b/src/gw_control/src/prov/src/gw_api_proxy/src/gw_api_proxy_svc.c
@@ -833,6 +833,12 @@ _proxy_docsis_getdsofdmchantable_1 (void  *argp, void *result, struct svc_req *r
 	return (proxy_docsis_getdsofdmchantable_1_svc(result, rqstp));
 }
 
+int
+_proxy_set_arm_inangoplug_enable_1 (void  *argp, void *result, struct svc_req *rqstp)
+{
+	return (proxy_set_arm_inangoplug_enable_1_svc(result, rqstp));
+}
+
 static void
 cgm_gw_api_proxy_prog_number_1(struct svc_req *rqstp, register SVCXPRT *transp)
 {
@@ -1009,6 +1015,7 @@ cgm_gw_api_proxy_prog_number_1(struct svc_req *rqstp, register SVCXPRT *transp)
 		PROXY_STATUS proxy_docsis_shut_down_1_res;
 		PROXY_STATUS proxy_cm_hal_gw_mode_change_1_res;
 		RPC_PROXY_DOCSIF31_CM_DS_OFDM_CHAN proxy_docsis_getdsofdmchantable_1_res;
+		PROXY_STATUS proxy_set_arm_inangoplug_enable_1_res;
 	} result;
 	bool_t retval;
 	xdrproc_t _xdr_argument, _xdr_result;
@@ -1795,6 +1802,12 @@ cgm_gw_api_proxy_prog_number_1(struct svc_req *rqstp, register SVCXPRT *transp)
 		local = (bool_t (*) (char *, void *,  struct svc_req *))_proxy_docsis_getdsofdmchantable_1;
 		break;
 
+	case PROXY_SET_ARM_INANGOPLUG_ENABLE:
+		_xdr_argument = (xdrproc_t) xdr_void;
+		_xdr_result = (xdrproc_t) xdr_PROXY_STATUS;
+		local = (bool_t (*) (char *, void *,  struct svc_req *))_proxy_set_arm_inangoplug_enable_1;
+		break;
+
 	default:
 		svcerr_noproc (transp);
 		return;
