diff --git a/rdkb_debug.ini b/rdkb_debug.ini
index d28836e..b4bce3f 100644
--- a/rdkb_debug.ini
+++ b/rdkb_debug.ini
@@ -61,3 +61,4 @@ LOG.RDK.SPDTST = ALL FATAL ERROR WARNING NOTICE INFO DEBUG TRACE
 LOG.RDK.BLE = ALL FATAL ERROR WARNING NOTICE INFO DEBUG TRACE
 LOG.RDK.WANAGENT = ALL FATAL ERROR WARNING NOTICE INFO DEBUG TRACE
 LOG.RDK.TELCOVOIPAGENT = ALL FATAL ERROR WARNING NOTICE INFO DEBUG TRACE
+LOG.RDK.INANGOPLUG = ALL FATAL ERROR WARNING NOTICE INFO DEBUG TRACE
diff --git a/rdkb_log4crc b/rdkb_log4crc
index 49b3e4b..ae2bb76 100755
--- a/rdkb_log4crc
+++ b/rdkb_log4crc
@@ -103,6 +103,7 @@
     <rollingpolicy name="BLE_rollingpolicy" type="sizewin" maxsize="2097152" maxnum="2"/>
     <rollingpolicy name="WANAGENT_rollingpolicy" type="sizewin" maxsize="2097152" maxnum="2"/>
     <rollingpolicy name="TELCOVOIPAGENT_rollingpolicy" type="sizewin" maxsize="2097152" maxnum="2"/>
+    <rollingpolicy name="INANGOPLUG_rollingpolicy" type="sizewin" maxsize="2097152" maxnum="2"/>
 
  <!--   <appender name="stderr" type="stream_env" layout="dated"/>
     <appender name="syslog" type="syslog" layout="basic"/>
@@ -149,6 +150,7 @@
     <appender name="RI_BLErollingfileappender" type="rollingfile" logdir="/rdklogs/logs/" prefix="Blelog.txt" layout="comcast_dated" rollingpolicy="BLE_rollingpolicy"/>
     <appender name="RI_WANAGENTrollingfileappender" type="rollingfile" logdir="/rdklogs/logs/" prefix="WANAGENTLog.txt" layout="comcast_dated" rollingpolicy="WANAGENT_rollingpolicy"/>
     <appender name="RI_TELCOVOIPAGENTrollingfileappender" type="rollingfile" logdir="/rdklogs/logs/" prefix="TELCOVOIPAGENTLog.txt" layout="comcast_dated" rollingpolicy="TELCOVOIPAGENT_rollingpolicy"/>
+    <appender name="RI_INANGOPLUGrollingfileappender" type="rollingfile" logdir="/rdklogs/logs/" prefix="InangoplugLog.txt" layout="comcast_dated" rollingpolicy="INANGOPLUG_rollingpolicy"/>
 
     <!-- <appender name="stdout" type="stream_env" layout="comcast_dated"/> -->
 
@@ -227,6 +229,8 @@
     <category name="RI.BLE" priority="debug" appender="RI_BLErollingfileappender"/>
     <category name="RI.WANAGENT" priority="debug" appender="RI_WANAGENTrollingfileappender"/>
     <category name="RI.TELCOVOIPAGENT" priority="debug" appender="RI_TELCOVOIPAGENTrollingfileappender"/>
+    <category name="RI.INANGOPLUG" priority="debug" appender="RI_INANGOPLUGrollingfileappender"/>
+
     <!--  
         Logging categories available to the Platform are listed below
         
@@ -354,4 +358,6 @@
 
      <category name="RI.Stack.BLE" priority="debug" appender="RI_BLErollingfileappender"/>
         <category name="RI.Stack.LOG.RDK.BLE" priority="debug" appender="RI_BLErollingfileappender"/>
+     <category name="RI.Stack.INANGOPLUG" priority="debug" appender="RI_INANGOPLUGrollingfileappender"/>
+     <category name="RI.Stack.LOG.RDK.INANGOPLUG" priority="debug" appender="RI_INANGOPLUGrollingfileappender"/>
 </log4c>
