diff --git a/rdkb_debug.ini b/rdkb_debug.ini
index e09b0be..91248b7 100644
--- a/rdkb_debug.ini
+++ b/rdkb_debug.ini
@@ -71,3 +71,4 @@ LOG.RDK.XTMAGENT = ALL FATAL ERROR WARNING NOTICE INFO DEBUG TRACE
 LOG.RDK.CPUPROCANALYZER = ALL FATAL ERROR WARNING NOTICE INFO DEBUG
 LOG.RDK.HWST = ERROR NOTICE
 LOG.RDK.OVSAGENT = ALL FATAL ERROR WARNING NOTICE INFO DEBUG TRACE
+LOG.RDK.INANGOPLUG = ALL FATAL ERROR WARNING NOTICE INFO DEBUG TRACE
diff --git a/rdkb_log4crc b/rdkb_log4crc
index 9cf2b2d..be4d10b 100644
--- a/rdkb_log4crc
+++ b/rdkb_log4crc
@@ -113,6 +113,7 @@
     <rollingpolicy name="CPUPROCANALYZER_rollingpolicy" type="sizewin" maxsize="2097152" maxnum="2"/>
     <rollingpolicy name="HWST_rollingpolicy" type="sizewin" maxsize="2097152" maxnum="2"/>
     <rollingpolicy name="OVSAGENT_rollingpolicy" type="sizewin" maxsize="2097152" maxnum="2"/>
+    <rollingpolicy name="INANGOPLUG_rollingpolicy" type="sizewin" maxsize="2097152" maxnum="2"/>
 
  <!--   <appender name="stderr" type="stream_env" layout="dated"/>
     <appender name="syslog" type="syslog" layout="basic"/>
@@ -169,6 +170,7 @@
     <appender name="RI_CPUPROCANALYZERrollingfileappender" type="rollingfile" logdir="/rdklogs/logs/" prefix="CPUPROCANALYZERlog.txt" layout="comcast_dated" rollingpolicy="CPUPROCANALYZER_rollingpolicy"/>
     <appender name="RI_HWSTrollingfileappender" type="rollingfile" logdir="/rdklogs/logs/" prefix="HWSTLog.txt" layout="comcast_dated" rollingpolicy="HWST_rollingpolicy"/>
     <appender name="RI_OVSAGENTrollingfileappender" type="rollingfile" logdir="/rdklogs/logs/" prefix="OvsAgentLog.txt" layout="comcast_dated" rollingpolicy="OVSAGENT_rollingpolicy"/>
+    <appender name="RI_INANGOPLUGrollingfileappender" type="rollingfile" logdir="/rdklogs/logs/" prefix="InangoplugLog.txt" layout="comcast_dated" rollingpolicy="INANGOPLUG_rollingpolicy"/>
 
     <!-- <appender name="stdout" type="stream_env" layout="comcast_dated"/> -->
 
@@ -257,6 +259,7 @@
     <category name="RI.CPUPROCANALYZER" priority="debug" appender="RI_CPUPROCANALYZERrollingfileappender"/>
     <category name="RI.HWST" priority="debug" appender="RI_HWSTrollingfileappender"/>
     <category name="RI.OVSAGENT" priority="debug" appender="RI_OVSAGENTrollingfileappender"/>
+    <category name="RI.INANGOPLUG" priority="debug" appender="RI_INANGOPLUGrollingfileappender"/>
 
     <!--
         Logging categories available to the Platform are listed below
@@ -409,4 +412,7 @@
 
      <category name="RI.Stack.OVSAGENT" priority="debug" appender="RI_OVSAGENTrollingfileappender"/>
      <category name="RI.Stack.LOG.RDK.OVSAGENT" priority="debug" appender="RI_OVSAGENTrollingfileappender"/>
+
+     <category name="RI.Stack.INANGOPLUG" priority="debug" appender="RI_INANGOPLUGrollingfileappender"/>
+     <category name="RI.Stack.LOG.RDK.INANGOPLUG" priority="debug" appender="RI_INANGOPLUGrollingfileappender"/>
 </log4c>
