diff --git a/scripts/LogAgent.xml b/scripts/LogAgent.xml
index 7ea023c..5b6d488 100644
--- a/scripts/LogAgent.xml
+++ b/scripts/LogAgent.xml
@@ -350,6 +350,18 @@
             <syntax>bool</syntax>
             <writable>true</writable>
         </parameter>	    
+        <parameter>
+            <name>X_INANGO_Inangoplug_LogLevel</name>
+            <type>unsignedInt</type>
+            <syntax>uint32</syntax>
+            <writable>true</writable>
+        </parameter>
+        <parameter>
+            <name>X_INANGO_Inangoplug_LoggerEnable</name>
+            <type>boolean</type>
+            <syntax>bool</syntax>
+            <writable>true</writable>
+        </parameter>
       </parameters>
     </object>
  </objects>
diff --git a/source/LogComponent/cosa_apis_logagentplugin.c b/source/LogComponent/cosa_apis_logagentplugin.c
index c449687..64b5b43 100644
--- a/source/LogComponent/cosa_apis_logagentplugin.c
+++ b/source/LogComponent/cosa_apis_logagentplugin.c
@@ -41,6 +41,7 @@
 #define DSLAGENT_PROC_NAME "dslagent"
 #define XTMAGENT_PROC_NAME "xtmagent"
 #define VLANAGENT_PROC_NAME "vlanagent"
+#define INANGOPLUG_PROC_NAME "inangoplug_component"
 
 #if defined (FEATURE_RDKB_WAN_MANAGER)
 #define VLANMANAGER_PROC_NAME "VlanManager"
@@ -607,6 +608,20 @@ LogAgent_GetParamUlongValue
             *puLong  = VLANAGENT_RDKLogLevel;
             return TRUE;
         }
+        if (AnscEqualString(ParamName, "X_INANGO_Inangoplug_LogLevel", TRUE))
+        {
+	       	syscfg_get( NULL, "X_INANGO_Inangoplug_LogLevel", buf, sizeof(buf));
+        	if( buf[0] != '\0' )
+        	{
+        		*puLong  = (ULONG )atoi(buf);
+                	INANGOPLUG_RDKLogLevel = *puLong;
+        	}
+        	else
+        	{
+        		AnscTraceWarning(("Error in syscfg_get for INANGOPLUG_LogLevel\n"));
+        	}
+        	return TRUE;
+        }
         return FALSE;
 }
  /**********************************************************************  
@@ -1351,6 +1366,25 @@ LogAgent_SetParamUlongValue
         SendSignal(VLANAGENT_PROC_NAME);
         return TRUE;
     }
+    if (AnscEqualString(ParamName, "X_INANGO_Inangoplug_LogLevel", TRUE))
+    {
+        char buf[8]={ 0 };
+        INANGOPLUG_RDKLogLevel = uValue;
+        snprintf(buf,sizeof(buf),"%lu",uValue);
+        if (syscfg_set(NULL, "X_INANGO_Inangoplug_LogLevel", buf) != 0)
+        {
+            AnscTraceWarning(("syscfg_set failed\n"));
+        }
+        else
+        {
+            if (syscfg_commit() != 0)
+            {
+                AnscTraceWarning(("syscfg_commit failed\n"));
+            }
+        }
+        SendSignal(INANGOPLUG_PROC_NAME);
+        return TRUE;
+    }
 
 	return FALSE;
 }
@@ -1929,6 +1963,20 @@ LogAgent_GetParamBoolValue
         *pBool  = VLANAGENT_RDKLogEnable;
         return TRUE;
     }    
+    if (AnscEqualString(ParamName, "X_INANGO_Inangoplug_LoggerEnable", TRUE))
+    {
+        syscfg_get( NULL, "X_INANGO_Inangoplug_LoggerEnable", buf, sizeof(buf));
+        if( buf[0] != '\0' )
+        {
+            	*pBool = (BOOL)atoi(buf);
+            	INANGOPLUG_RDKLogEnable = *pBool;
+        }
+        else
+        {
+            	AnscTraceWarning(("Error in syscfg_get for INANGOPLUG_LoggerEnable\n"));
+        }
+        return TRUE;
+    }
     /* AnscTraceWarning(("Unsupported parameter '%s'\n", ParamName)); */
     return FALSE;
 }
@@ -2560,6 +2608,25 @@ LogAgent_SetParamBoolValue
         SendSignal(VLANAGENT_PROC_NAME);
         return TRUE;
     }
+    if (AnscEqualString(ParamName, "X_INANGO_Inangoplug_LoggerEnable", TRUE))
+    {
+        char buf[8];
+        INANGOPLUG_RDKLogEnable = bValue;
+        snprintf(buf,sizeof(buf),"%d",bValue);
+        if (syscfg_set(NULL, "X_INANGO_Inangoplug_LoggerEnable", buf) != 0)
+        {
+            AnscTraceWarning(("syscfg_set failed\n"));
+        }
+        else
+        {
+            if (syscfg_commit() != 0)
+            {
+                AnscTraceWarning(("syscfg_commit failed\n"));
+            }
+        }
+        SendSignal(INANGOPLUG_PROC_NAME);
+        return TRUE;
+    }
     /* AnscTraceWarning(("Unsupported parameter '%s'\n", ParamName)); */
     return FALSE;
 }
diff --git a/source/LogComponent/ssp_main.c b/source/LogComponent/ssp_main.c
index fcb853d..dc684fc 100644
--- a/source/LogComponent/ssp_main.c
+++ b/source/LogComponent/ssp_main.c
@@ -508,6 +508,16 @@ void ReadLogInfo()
         {
             XTMAGENT_RDKLogEnable = (BOOL)atoi(buf);
         }
+        syscfg_get( NULL, "X_INANGO_Inangoplug_LogLevel", buf, sizeof(buf));
+        if( buf != NULL )
+        {
+            INANGOPLUG_RDKLogLevel = atoi(buf);
+        }
+        syscfg_get( NULL, "X_INANGO_Inangoplug_LoggerEnable", buf, sizeof(buf));
+        if( buf != NULL )
+        {
+            INANGOPLUG_RDKLogEnable = (BOOL)atoi(buf);
+        }
 }
 
 int main(int argc, char* argv[])
