diff --git a/source/debug_api/ansc_debug.c b/source/debug_api/ansc_debug.c
index 12fd053f..16bb1ea3 100644
--- a/source/debug_api/ansc_debug.c
+++ b/source/debug_api/ansc_debug.c
@@ -158,6 +158,8 @@ volatile unsigned int TELCOVOICEMANAGER_RDKLogLevel = 4;
 volatile BOOL PPPMANAGER_RDKLogEnable = TRUE;
 volatile unsigned int PPPMANAGER_RDKLogLevel = 4;
 #endif //FEATURE_RDKB_WAN_MANAGER
+volatile BOOL INANGOPLUG_RDKLogEnable = TRUE;
+volatile unsigned int INANGOPLUG_RDKLogLevel = 4;
 /**********************************************************************
                     VARIABLES FOR TRACE LEVEL
 **********************************************************************/
@@ -495,6 +497,12 @@ void CcspTraceLogAPI(char *fileName, char *pComponentName, int level, const char
             LogLevel = PPPMANAGER_RDKLogLevel;
             LogEnable = PPPMANAGER_RDKLogEnable;
         }	
+        else if(!strcmp(pComponentName,"com.cisco.spvtg.ccsp.inangoplugcomponent"))
+        {
+            ComponentName= "LOG.RDK.INANGOPLUG";
+            LogLevel = INANGOPLUG_RDKLogLevel;
+            LogEnable = INANGOPLUG_RDKLogEnable;
+        }
 #endif // FEATURE_RDKB_WAN_MANAGER
         else
         {
diff --git a/source/debug_api/include/ccsp_trace.h b/source/debug_api/include/ccsp_trace.h
index d21214a0..140ceea8 100644
--- a/source/debug_api/include/ccsp_trace.h
+++ b/source/debug_api/include/ccsp_trace.h
@@ -173,6 +173,9 @@ extern volatile BOOL PPPMANAGER_RDKLogEnable;
 
 extern volatile unsigned int BLE_RDKLogLevel;
 extern volatile BOOL BLE_RDKLogEnable;
+
+extern volatile unsigned int INANGOPLUG_RDKLogLevel;
+extern volatile BOOL INANGOPLUG_RDKLogEnable;
 /*
  *  Whether a debug trace is output depends on the following factors:
  *      1) the trace level passed in the trace statement is higher (smaller value)
