Index: git/source/scripts/init/system/utopia_init.sh
===================================================================
--- git.orig/source/scripts/init/system/utopia_init.sh
+++ git/source/scripts/init/system/utopia_init.sh
@@ -538,6 +538,30 @@ echo "[utopia][init] SEC: Syscfg stored
 syscfg unset UpdateNvram
 syscfg commit
 
+OVS_ENABLE=`syscfg get CONFIG_OVS_INFRASTRUCTURE_ENABLE`
+loopBreakCounter=0
+
+touch /tmp/syscfg_inited
+
+if [ "$OVS_ENABLE" == "true" ]
+then
+    while [ ! -f /tmp/inangoplug_started ]
+    do
+        if [ "$loopBreakCounter" -ne 30 ]; then
+            echo "[utopia][init] inangoplug is not ready yet..." > /dev/console
+            loopBreakCounter=$((loopBreakCounter+1))
+            sleep 1
+        else
+            break
+        fi
+    done
+    if [ -f /tmp/inangoplug_started ]; then
+        echo "[utopia][init] inangoplug is enabled and started successful" > /dev/console
+    else
+        echo "[utopia][init] inangoplug is enabled but not started properly" > /dev/console
+    fi
+fi
+
 #Added log to check the DHCP range corruption after system defaults applied.
 lan_ipaddr=`syscfg get lan_ipaddr`
 lan_netmask=`syscfg get lan_netmask`
