Index: git/scripts/LogAgent.xml
===================================================================
--- git.orig/scripts/LogAgent.xml
+++ git/scripts/LogAgent.xml
@@ -350,6 +350,18 @@
             <syntax>bool</syntax>
             <writable>true</writable>
         </parameter>	    
+        <parameter>
+            <name>X_INANGO_Inangoplug_LogLevel</name>
+            <type>unsignedInt</type>
+            <syntax>uint32</syntax>
+            <writable>true</writable>
+        </parameter>
+        <parameter>
+            <name>X_INANGO_Inangoplug_LoggerEnable</name>
+            <type>boolean</type>
+            <syntax>bool</syntax>
+            <writable>true</writable>
+        </parameter>
       </parameters>
     </object>
  </objects>
Index: git/source/LogComponent/cosa_apis_logagentplugin.c
===================================================================
--- git.orig/source/LogComponent/cosa_apis_logagentplugin.c
+++ git/source/LogComponent/cosa_apis_logagentplugin.c
@@ -38,6 +38,7 @@
 #define MESH_PROC_NAME "meshAgent"
 #define ETHAGENT_PROC_NAME "CcspEthAgent"
 #define TELCOVOIPAGENT_PROC_NAME "telcovoip_agent"
+#define INANGOPLUG_PROC_NAME "inangoplug_component"
 #define DSLAGENT_PROC_NAME "dslagent"
 #define XTMAGENT_PROC_NAME "xtmagent"
 #define VLANAGENT_PROC_NAME "vlanagent"
@@ -607,6 +608,20 @@ LogAgent_GetParamUlongValue
             *puLong  = VLANAGENT_RDKLogLevel;
             return TRUE;
         }
+        if (AnscEqualString(ParamName, "X_INANGO_Inangoplug_LogLevel", TRUE))
+        {
+            syscfg_get( NULL, "X_INANGO_Inangoplug_LogLevel", buf, sizeof(buf));
+            if( buf[0] != '\0' )
+            {
+                *puLong  = (ULONG )atoi(buf);
+                INANGOPLUG_RDKLogLevel = *puLong;
+            }
+            else
+            {
+                AnscTraceWarning(("Error in syscfg_get for INANGOPLUG_LogLevel\n"));
+            }
+            return TRUE;
+        }
         return FALSE;
 }
  /**********************************************************************  
@@ -1352,6 +1367,25 @@ LogAgent_SetParamUlongValue
         return TRUE;
     }
 
+    if (AnscEqualString(ParamName, "X_INANGO_Inangoplug_LogLevel", TRUE))
+    {
+        char buf[8]={ 0 };
+        INANGOPLUG_RDKLogLevel = uValue;
+        snprintf(buf,sizeof(buf),"%lu",uValue);
+        if (syscfg_set(NULL, "X_INANGO_Inangoplug_LogLevel", buf) != 0)
+        {
+            AnscTraceWarning(("syscfg_set failed\n"));
+        }
+        else
+        {
+            if (syscfg_commit() != 0)
+            {
+                AnscTraceWarning(("syscfg_commit failed\n"));
+            }
+        }
+        SendSignal(INANGOPLUG_PROC_NAME);
+        return TRUE;
+    }
 	return FALSE;
 }
      
@@ -1928,7 +1962,22 @@ LogAgent_GetParamBoolValue
     {
         *pBool  = VLANAGENT_RDKLogEnable;
         return TRUE;
-    }    
+    }
+
+    if (AnscEqualString(ParamName, "X_INANGO_Inangoplug_LoggerEnable", TRUE))
+    {
+        syscfg_get( NULL, "X_INANGO_Inangoplug_LoggerEnable", buf, sizeof(buf));
+        if( buf[0] != '\0' )
+        {
+            *pBool = (BOOL)atoi(buf);
+             INANGOPLUG_RDKLogEnable = *pBool;
+        }
+        else
+        {
+            AnscTraceWarning(("Error in syscfg_get for INANGOPLUG_LoggerEnable\n"));
+        }
+        return TRUE;
+    }
     /* AnscTraceWarning(("Unsupported parameter '%s'\n", ParamName)); */
     return FALSE;
 }
@@ -2560,6 +2609,26 @@ LogAgent_SetParamBoolValue
         SendSignal(VLANAGENT_PROC_NAME);
         return TRUE;
     }
+    
+    if (AnscEqualString(ParamName, "X_INANGO_Inangoplug_LoggerEnable", TRUE))
+    {
+        char buf[8];
+        INANGOPLUG_RDKLogEnable = bValue;
+        snprintf(buf,sizeof(buf),"%d",bValue);
+        if (syscfg_set(NULL, "X_INANGO_Inangoplug_LoggerEnable", buf) != 0)
+        {
+            AnscTraceWarning(("syscfg_set failed\n"));
+        }
+        else
+        {
+            if (syscfg_commit() != 0)
+            {
+                AnscTraceWarning(("syscfg_commit failed\n"));
+            }
+        }
+        SendSignal(INANGOPLUG_PROC_NAME);
+        return TRUE;
+    }
     /* AnscTraceWarning(("Unsupported parameter '%s'\n", ParamName)); */
     return FALSE;
 }
Index: git/source/LogComponent/ssp_main.c
===================================================================
--- git.orig/source/LogComponent/ssp_main.c
+++ git/source/LogComponent/ssp_main.c
@@ -508,6 +508,16 @@ void ReadLogInfo()
         {
             XTMAGENT_RDKLogEnable = (BOOL)atoi(buf);
         }
+        syscfg_get( NULL, "X_INANGO_Inangoplug_LogLevel", buf, sizeof(buf));
+        if( buf != NULL )
+        {
+            INANGOPLUG_RDKLogLevel = atoi(buf);
+        }
+        syscfg_get( NULL, "X_INANGO_Inangoplug_LoggerEnable", buf, sizeof(buf));
+        if( buf != NULL )
+        {
+            INANGOPLUG_RDKLogEnable = (BOOL)atoi(buf);
+        }
 }
 
 int main(int argc, char* argv[])
